satpy/readers/seviri_l1b_native.py

```diff
@@ -288,14 +288,16 @@ def get_area_def(self, dataset_id):
         pdict['ssp_lon'] = self.mda['projection_parameters']['ssp_longitude']
 
         if dataset_id['name'] == 'HRV':
-            pdict['a_name'] = 'geos_seviri_hrv'
-            pdict['a_desc'] = 'SEVIRI high resolution channel area'
+            res = 1.0
             pdict['p_id'] = 'seviri_hrv'
         else:
-            pdict['a_name'] = 'geos_seviri_visir'
-            pdict['a_desc'] = 'SEVIRI low resolution channel area'
+            res = 3.0
             pdict['p_id'] = 'seviri_visir'
 
+        service_mode = get_service_mode(pdict['ssp_lon'])
+        pdict['a_name'] = 'msg_seviri_%s_%.0fkm' % (service_mode['name'], res)
+        pdict['a_desc'] = 'SEVIRI %s area definition with %.0f km resolution' % (service_mode['desc'], res)
```

Comment:
Thanks for contributing to https://github.com/pytroll/satpy/issues/1248 ! We could have the formatting of the name and description standardized in a separate function in `seviri_base`, to be accessible by all readers. But that's for another PR :)

Reply:
Care to make an issue for this?